### PR TITLE
CASSANDRASC-43 Add Schema API

### PR DESCRIPTION
**Pull request by frankgh**
This commit introduces the Schema API. Two new endpoints are added:

- /api/v1/schema/keyspaces This endpoint returns the SchemaResponse with the full schema for all keyspaces.

- /api/v1/schema/keyspaces/:keyspace This endpoint returns the SchemaResponse with the full schema for the requested keyspace.

### CODE REVIEW

File: src/test/resources/schema/test_keyspace_schema.cql [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+
+CREATE TYPE testkeyspace.cidr_range (
+    id uuid,
+    start inet,
+    end inet,
+    description text
+);
+
+CREATE TABLE testkeyspace.testtable (
+                                        partition_key_1 uuid,
```
Comment (yifan-c) [inline src/test/resources/schema/test_keyspace_schema.cql]: nit: can be nicely indented the same as `CREATE TYPE`

File: common/src/main/java/org/apache/cassandra/sidecar/common/data/SchemaResponse.java [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+    {
+        this.keyspace = Objects.requireNonNull(keyspace, "keyspace must be non-null");
+        this.schema = Objects.requireNonNull(schema, "schema must be non-null");
+    }
+
+    /**
+     * @return the name of the Cassandra keyspace
+     */
+    @JsonInclude(JsonInclude.Include.NON_NULL)
+    public String getKeyspace()
```
Comment (dineshjoshi) [inline common/src/main/java/org/apache/cassandra/sidecar/common/data/SchemaResponse.java]: `getKeyspace()` -> `keyspace()`

File: common/src/main/java/org/apache/cassandra/sidecar/common/data/SchemaResponse.java [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+    {
+        return keyspace;
+    }
+
+    /**
+     * @return the string representing the schema for the response
+     */
+    public String getSchema()
```
Comment (dineshjoshi) [inline common/src/main/java/org/apache/cassandra/sidecar/common/data/SchemaResponse.java]: `getSchema()` -> `schema()`

File: common/src/main/java/org/apache/cassandra/sidecar/common/data/SchemaRequest.java [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+        super(keyspace, null, false);
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    public String toString()
+    {
+        return "SchemaRequest{" +
+               "keyspace='" + getKeyspace() + '\'' +
```
Comment (dineshjoshi) [inline common/src/main/java/org/apache/cassandra/sidecar/common/data/SchemaRequest.java]: Please drop the get before the getters...

**Review by dineshjoshi — CHANGES_REQUESTED** [review]
Minor changes required.

**Review by dineshjoshi — APPROVED** [review]
+1

**Comment by frankgh** [comment]
Closed via https://github.com/apache/cassandra-sidecar/commit/6e358acfce071cad16ac88c15dc2229bbb8a7944

**Pull request closed without merging**